builtin/providers/aws/resource_aws_s3_bucket_object_test.go

```diff
@@ -322,3 +340,18 @@ resource "aws_s3_bucket_object" "object" {
 }
 `, randInt, source, source)
 }
+
+func testAccAWSS3BucketObjectConfig_withKMSId(randInt int) string {
+	return fmt.Sprintf(`
+resource "aws_s3_bucket" "object_bucket_2" {
+	bucket = "tf-object-test-bucket-%d"
+}
+
+resource "aws_s3_bucket_object" "object" {
+	bucket = "${aws_s3_bucket.object_bucket_2.bucket}"
+	key = "test-key"
+	content = "stuff"
+	kms_key_id = "01961aed-d0b6-4ad3-9f7f-8264818ea611"
```

Comment:
Where is this key ID coming from?

Either we'll need to wait until https://github.com/hashicorp/terraform/pull/3928 is merged or introduce an environment variable for this.